test/val/val_limits_test.cpp

```diff
@@ -320,3 +321,71 @@ TEST_F(ValidateLimits, StructNestingDepthBad) {
       HasSubstr(
           "Structure Nesting Depth may not be larger than 255. Found 256."));
 }
+
+// clang-format off
+// Generates an SPIRV program with the given control flow nesting depth
+void GenerateSpirvProgramWithCfgNestingDepth(std::string& str, int depth) {
```

Comment:
This test is good as far as it goes.
It doesn't capture the issue of having a complex the old algorithm incorrectly calculating the depths of the nodes in a complex continue target that can be reached from an inner construct.  However, making that test case won't be very clear if you only check for the overall validator error message.

The validator has this issue in general: it really begs for exposing its internals so they can be better unit tested.  This is true for the dominance calculations, and now for the block depth calculations.

I made https://github.com/KhronosGroup/SPIRV-Tools/projects/1#card-1185526 and https://github.com/KhronosGroup/SPIRV-Tools/projects/1#card-1185539 for that refactoring-and-testing work.